### DISCUSSION TITLE
How to use ADS1115

### DISCUSSION THREAD

**Discussion by thiensu1122** [HefnySco/OBAL#11] [Q&A] [answered]
Hi HefnySco,

1) I know that ADS1115 can measure the voltage from 0-5v. But in my case, I use a 3S battery which means the max voltage is about 12.6V. Yet, I don't think the ADS1115 can measure the voltage directly so is it ok to use a battery divider? (it may be a silly question but I am not a electric engineer so it is a bit scary for me to work with the high ampere of LIPO battery)

2) Furthermore I have no idea how to measure the Ampare with this module. 

Can you share with me some ideas?

**Comment by HefnySco** [comment, marked as answer]
The easiest way is to use a power module and use the voltage and amp wires to connect them to ADS1115.

The hardway... not very hard.... is to use a voltage divider for the battery as you cannot use 12.6 as input to this ic otherwise it will die.
for ampare you use an ampare sensor that gives you voltage as output and then you connect this to ADS1115.
These functions is what a Power Module Cable does. SO a power.

![image](https://user-images.githubusercontent.com/4184373/163359147-f633de91-6e3a-44fb-bb38-1eb0e69e692f.png)

**Comment by HefnySco** [comment]
There is a fast hack you can do.
Connect each cell to ADS1115 A1,A2,A3
and define 
BATT_MONITOR  "Voltage & Current" with A0 as current and A1 as voltage
Then define BATT2_Monitor as voltage only with A2 as voltage pin sensor.
and define BATT3_Monitor as voltage only with A3 as voltage pin sensor.
Now you can connect cell 1,2,3 to A1,2,3 and read the voltage.
For current you still need a current sensor module connected to A0 - unless you do not need it then you can ignore it-.

// sensor 1 as voltage & current sensor
BATT_MONITOR   4
BATT_VOLT_PIN 1
BATT_CURR_PIN 0
BATT_VOLT_MULT	1.23

// sensor 2 as voltage sensor
BATT2_MONITOR	3
BATT2_VOLT_PIN 2
BATT2_VOLT_MULT  0.999

// sensor 3 as voltage sensor
BATT3_MONITOR	3
BATT3_VOLT_PIN 3
BATT3_VOLT_MULT  0.999

**Comment by HefnySco** [comment]
If you want  ignore current and use up to 4 cell directly you can use:

// sensor 1 as voltage sensor
BATT3_MONITOR 3
BATT3_VOLT_PIN 0
BATT3_VOLT_MULT 0.999


// sensor 1 as voltage sensor
BATT_MONITOR 3
BATT_VOLT_PIN 1
BATT_VOLT_MULT 0.999

// sensor 3 as voltage sensor
BATT3_MONITOR 3
BATT3_VOLT_PIN 2
BATT3_VOLT_MULT 0.999


// sensor 4 as voltage sensor
BATT4_MONITOR 3
BATT4_VOLT_PIN 3
BATT4_VOLT_MULT 0.999
